dotnet/src/SemanticKernel/CoreSkills/TextMemorySkill.cs

```diff
@@ -37,49 +38,96 @@ public class TextMemorySkill
+
+        var memory = await context.Memory.GetAsync(collection, key);
+        return memory != null ? memory.Text : string.Empty;
+    }
+
+    /// <summary>
+    /// Recall an idea - a set of related memories
+    /// </summary>
+    /// <example>
+    /// SKContext["input"] = "what is the capital of France?"
+    /// {{memory.recallidea $input }} => "Paris"
+    /// </example>
+    /// <param name="ask">The idea to retrieve</param>
+    /// <param name="context">Contains the 'collection' to search for the idea and 'relevance' score</param>
+    [SKFunction("Recall an idea from the long term memory")]
+    [SKFunctionName("RecallIdea")]
+    [SKFunctionInput(Description = "The idea to recall")]
+    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection where to search for the idea", DefaultValue = DefaultCollection)]
     [SKFunctionContextParameter(Name = RelevanceParam, Description = "The relevance score, from 0.0 to 1.0, where 1.0 means perfect match",
         DefaultValue = DefaultRelevance)]
-    public async Task<string> RecallAsync(string ask, SKContext context)
+    [SKFunctionContextParameter(Name = LimitParam, Description = "The maximum number of relevant memories to recall", DefaultValue = DefaultLimit)]
+    [SKFunctionContextParameter(Name = JoinParam, Description = "String used to separate multiple memories", DefaultValue = DefaultJoiner)]
+    public async Task<string> RecallIdeaAsync(string ask, SKContext context)
     {
         var collection = context.Variables.ContainsKey(CollectionParam) ? context[CollectionParam] : DefaultCollection;
-        Verify.NotEmpty(collection, "Memory collection not defined");
+        Verify.NotEmpty(collection, "Memories collection not defined");
 
         var relevance = context.Variables.ContainsKey(RelevanceParam) ? context[RelevanceParam] : DefaultRelevance;
         if (string.IsNullOrWhiteSpace(relevance)) { relevance = DefaultRelevance; }
 
-        context.Log.LogTrace("Searching memory for '{0}', collection '{1}', relevance '{2}'", ask, collection, relevance);
+        var limit = context.Variables.ContainsKey(LimitParam) ? context[LimitParam] : LimitParam;
+        if (string.IsNullOrWhiteSpace(relevance)) { relevance = DefaultLimit; }
+
+        context.Log.LogTrace("Searching memories for '{0}', collection '{1}', relevance '{2}'", ask, collection, relevance);
 
         // TODO: support locales, e.g. "0.7" and "0,7" must both work
-        MemoryQueryResult? memory = await context.Memory
-            .SearchAsync(collection, ask, limit: 1, minRelevanceScore: float.Parse(relevance, CultureInfo.InvariantCulture))
-            .FirstOrDefaultAsync();
+        IAsyncEnumerable<MemoryQueryResult> memories = context.Memory
+            .SearchAsync(collection, ask, limit: int.Parse(limit, CultureInfo.InvariantCulture), minRelevanceScore: float.Parse(relevance, CultureInfo.InvariantCulture));
```

Comment:
Should we await here? I'm not super familiar with IAsyncEnumerable. I feel like we should, or remove the async from the method signature.

Reply:
IAsyncEnumerables are bizarre. I think I can remove async from the method.

Reply:
Yeah, just played around with this more. Fun stuff. removing async sounds right!